mat64/doc.go

```diff
@@ -112,7 +112,34 @@
 // For convenience, a matrix may be used as both a receiver and as an input, i.e.
 //  a.Pow(a, 6)
 //  v.SolveVec(a.T(), v)
-// though in many cases this will cause an allocation. However, mat64 cannot
-// detect arbitrary overlap, and so care should be taken when using matrix views.
+// though in many cases this will cause an allocation (see Element Aliasing).
 // An exception to this rule is Copy, which does not allow a.Copy(a.T()).
+//
+// Element Aliasing
+//
+// Most methods in mat64 modify receiver data. It is forbidden for the modified
+// data region of the receiver to overlap the used data area of the input
+// arguments. The exception to this rule is when the method receiver is equal to one
+// of the input arguments, as in the a.Pow(a, 6) call above, or its implicit transpose.
+//
+// If you need to read this section to understand the behavior of your program,
+// you are being clever. Don't be clever. If you must be clever, blas64 and
+// lapack64 may be used to call the behavior directly.
+//
+// When data is shared between the receiver and input, mat64 will attempt to detect
```

Comment:
> On Oct 18, 2015, at 6:29 PM, Dan Kortschak <notifications@github.com> wrote:
> 
> In mat64/doc.go <https://github.com/gonum/matrix/pull/263#discussion_r42329418>:
> 
> >  // An exception to this rule is Copy, which does not allow a.Copy(a.T()).
> > +//
> > +// Element Aliasing
> > +//
> > +// Most methods in mat64 modify receiver data. It is forbidden for the modified
> > +// data region of the receiver to overlap the used data area of the input
> > +// arguments. The exception to this rule is when the method receiver is equal to one
> > +// of the input arguments, as in the a.Pow(a, 6) call above, or its implicit transpose.
> > +//
> > +// If you need to read this section to understand the behavior of your program,
> > +// you are being clever. Don't be clever. If you must be clever, blas64 and
> > +// lapack64 may be used to call the behavior directly.
> > +//
> > +// When data is shared between the receiver and input, mat64 will attempt to detect
> mat64 does not detect element overlap if the input does not implement a Raw method or if the Raw method differs from that of the receiver.
> 
> This seems redundant. Is it just there to provide emphasis for the "Correct method behavior cannot be guaranteed if there is overlap" part?
> 
> 

Yes, basically. The idea is “We know you’re wrong if X. We guarantee you’re fine if Y. You’re wrong but we don’t tell you if Z"
> —
> Reply to this email directly or view it on GitHub <https://github.com/gonum/matrix/pull/263/files#r42329418>.
>